modules/ttdBidAdapter.js

```diff
@@ -0,0 +1,602 @@
+        return false;
+      }
+      if (mediaTypesVideo.skip && !(mediaTypesVideo.skip === 0 || mediaTypesVideo.skip === 1)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.skip should be either 0 or 1');
+        return false;
+      }
+      if (mediaTypesVideo.skipmin && !utils.isInteger(mediaTypesVideo.skipmin)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.skipmin should be an integer');
+        return false;
+      }
+      if (mediaTypesVideo.skipafter && !utils.isInteger(mediaTypesVideo.skipafter)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.skipafter should be an integer');
+        return false;
+      }
+      if (mediaTypesVideo.minbitrate && !utils.isInteger(mediaTypesVideo.minbitrate)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.minbitrate should be an integer');
+        return false;
+      }
+      if (mediaTypesVideo.maxbitrate && !utils.isInteger(mediaTypesVideo.maxbitrate)) {
+        utils.logWarn(BIDDER_CODE + ': mediaTypes.video.maxbitrate should be an integer');
+        return false;
+      }
+    }
+
+    const pageCategories = utils.deepAccess(bid, 'params.categories');
+    if (pageCategories) {
+      if (!Array.isArray(pageCategories)) {
+        utils.logWarn(BIDDER_CODE + ': params.categories should be an array of string values');
+        return false;
+      }
+      for (let i = 0; i < pageCategories.length; i++) {
+        if (typeof pageCategories[i] !== 'string') {
+          utils.logWarn(BIDDER_CODE + ': params.categories should be an array of string values');
+          return false;
+        }
+      }
+    }
+
+    const keywords = utils.deepAccess(bid, 'params.keywords');
+    if (keywords) {
+      if (!Array.isArray(keywords)) {
+        utils.logWarn(BIDDER_CODE + ': params.keywords should be an array of string values');
+        return false;
+      }
+      for (let i = 0; i < keywords.length; i++) {
+        if (typeof keywords[i] !== 'string') {
+          utils.logWarn(BIDDER_CODE + ': params.keywords should be an array of string values');
+          return false;
+        }
+        if (keywords[i].length > 128) {
+          utils.logWarn(BIDDER_CODE + ': params.keywords must only contain strings that are 128 characters or less');
+          return false;
+        }
+      }
+      if (keywords.length > 10) {
+        utils.logWarn(BIDDER_CODE + ': params.keywords must can only contain a max of 10 keywords');
+        return false;
+      }
+    }
+
+    return true;
+  },
+
+  /**
+   * Make a server request from the list of BidRequests.
+   *
+   * @param {BidRequest[]} an array of validBidRequests
+   * @param {*} bidderRequest
+   * @return {ServerRequest} Info describing the request to the server.
+   */
+  buildRequests: function (validBidRequests, bidderRequest) {
+    utils.logInfo('validBidRequests:', JSON.stringify(validBidRequests))
+    utils.logInfo('bidderRequest:', JSON.stringify(bidderRequest))
```

Comment:
this seems to be used for development and can likely be removed

Reply:
removed all debug logging